fe/fe-core/src/main/java/com/starrocks/catalog/ExpressionRangePartitionInfo.java

```diff
@@ -116,7 +117,17 @@ public void gsonPostProcess() throws IOException {
             //  and partition by expr tables will use ExpressionRangePartitionInfoV2
             if (partitionNameColumnMap.containsKey(slotRef.getColumnName())) {
                 Column partitionColumn = partitionNameColumnMap.get(slotRef.getColumnName());
-                slotRef.setType(partitionColumn.getType());
+                if (slotRef.getType() == Type.INVALID) {
```

Comment:
why slotRef.getType() can be Type.INVALID?
can we guarantee that slotRef.type is always legal? so here just use Precondiation.check.

Reply:
Why do this?

`partitionRefTableExprs` is not serialized for `gson` and only serializes the expression-sql instead which I think is more compatible for upgrade/downgrade.

so need to recover it in `gsonPostProcess` phase, `expressionSql ` is only `parseSqlToExpr` and no analyzed and needs to check expression's type here to make it more compatible for further usage.

```
    @SerializedName(value = "partitionExprs")
    private List<GsonUtils.ExpressionSerializedObject> serializedPartitionExprs;
```